tests/pass/tree_borrows/cell-local-after-foreign.rs

```diff
@@ -0,0 +1,60 @@
+//@compile-flags: -Zmiri-tree-borrows
+
+// Perform foreign access to a Cell and then local access.
+// Takes advantage of two-phase borrows.
+
+use std::cell::Cell;
+
+pub trait Foo {
+    fn lw_fr(&mut self, val: i32);
+    fn lw_fw(&mut self, val: ());
+    fn lr_fw(&mut self, val: ());
+    fn lr_fr(&mut self, val: i32);
+}
+
+impl Foo for Cell<i32> {
+    // local write after foreign read
+    fn lw_fr(&mut self, val: i32) {
+        self.set(val);
+    }
+
+    // local write after foreign write
+    fn lw_fw(&mut self, _val: ()) {
+        self.set(100);
+    }
+
+    // local read after foreign write
+    fn lr_fw(&mut self, _val: ()) {
+        self.get();
+    }
+
+    // local read after foreign read
+    #[allow(path_statements)]
```

Comment:
Please don't allow the lint.

Just rename `val` to `_val` if you don't intend to use it.